static/app/views/performance/newTraceDetails/traceLinksNavigation/useFindLinkedTraces.ts

```diff
@@ -0,0 +1,167 @@
+import {useMemo} from 'react';
+
+import type {TraceItemResponseAttribute} from 'sentry/views/explore/hooks/useTraceItemDetails';
+import {useSpans} from 'sentry/views/insights/common/queries/useDiscover';
+
+import type {ConnectedTraceConnection} from './traceLinkNavigationButton';
+
+/**
+ * Find the next trace by looking using the spans endpoint to query for a trace
+ * linking to the current trace as its previous trace.
+ */
+export function useFindNextTrace({
+  direction,
+  attributes,
+  nextTraceEndTimestamp,
+  nextTraceStartTimestamp,
+}: {
+  attributes: TraceItemResponseAttribute[];
+  direction: ConnectedTraceConnection;
+  nextTraceEndTimestamp: number;
+  nextTraceStartTimestamp: number;
+}): {isLoading: boolean; id?: string; trace?: string} {
+  const {currentTraceId, currentSpanId, projectId, environment} = useMemo(() => {
+    let _currentTraceId: string | undefined;
+    let _currentSpanId: string | undefined;
+    let _projectId: number | undefined;
+    let _environment: string | undefined;
+
+    for (const a of attributes) {
+      if (a.name === 'trace' && a.type === 'str') {
+        _currentTraceId = a.value;
+      } else if (a.name === 'transaction.span_id' && a.type === 'str') {
+        _currentSpanId = a.value;
+      } else if (a.name === 'project_id' && a.type === 'int') {
+        _projectId = a.value;
+      } else if (a.name === 'environment' && a.type === 'str') {
+        _environment = a.value;
+      }
+    }
+    return {
+      currentTraceId: _currentTraceId,
+      currentSpanId: _currentSpanId,
+      projectId: _projectId,
+      environment: _environment,
+    };
+  }, [attributes]);
```

Comment:
(M - I'm not sure how much attention we give to objects in dependency arrays): As `attributes` is a non-primitive type in JS, the `useMemo` will probably create even a little performance overhead. As the object reference is always different, `useMemo` will probably be called on every render (and thus, not really memoize, if even if the array is the same).

There are different methods to account for this, I like this comment: https://github.com/facebook/react/issues/14476#issuecomment-471199055